pulp_ansible/tests/functional/api/collection/v3/test_collection_version_search.py

```diff
@@ -0,0 +1,1201 @@
+    def test_collection_version_search_by_not_deprecated(
+        self, galaxy_v3_default_search_api_client, search_specs
+    ):
+        """Get only CV's that are not deprecated."""
+        self._run_search(
+            galaxy_v3_default_search_api_client,
+            search_specs,
+            ["not spec.get('deprecated')"],
+            {"deprecated": False},
+        )
+
+    @pytest.mark.pulp_on_localhost
+    def test_collection_version_search_by_signed(
+        self, galaxy_v3_default_search_api_client, search_specs
+    ):
+        """Get only CV's that are signed."""
+        self._run_search(
+            galaxy_v3_default_search_api_client,
+            search_specs,
+            ["spec.get('signed') is True"],
+            {"signed": True},
+        )
+
+    @pytest.mark.pulp_on_localhost
+    def test_collection_version_search_by_not_signed(
+        self, galaxy_v3_default_search_api_client, search_specs
+    ):
+        """Get only CV's that are not signed."""
+        self._run_search(
+            galaxy_v3_default_search_api_client,
+            search_specs,
+            ["not spec.get('signed')"],
+            {"signed": False},
+        )
+
+
+def test_cross_repo_search_index_on_distribution_without_repository(
+    ansible_distro_api_client,
+    galaxy_v3_default_search_api_client,
+    gen_object_with_cleanup,
+):
+    """Make sure no indexes are created for a repo-less distribution."""
+    distro_data = gen_distribution()
+    distro = gen_object_with_cleanup(ansible_distro_api_client, distro_data)
+    dist_id = distro.pulp_href.split("/")[-2]
+    resp = galaxy_v3_default_search_api_client.list(limit=1000, distribution=[dist_id])
+    assert resp.meta.count == 0
+
+
+def test_cross_repo_search_index_on_distribution_with_repository_latest_version(
+    ansible_distro_api_client,
+    ansible_repo_api_client,
+    ansible_repo_version_api_client,
+    build_and_upload_collection,
+    galaxy_v3_default_search_api_client,
+    gen_object_with_cleanup,
+):
+    """Make sure indexes are created for a distribution that points at a repo version."""
+
+    pulp_repo = gen_object_with_cleanup(ansible_repo_api_client, {"name": str(uuid.uuid4())})
+    build_and_upload_collection(ansible_repo=pulp_repo)
+
+    # get the latest repo version ...
+    repository_versions = ansible_repo_version_api_client.list(pulp_repo.pulp_href)
+    repository_version_latest = repository_versions.results[0]
+
+    # make a distro that points only at the latest repo version ...
+    distro = gen_object_with_cleanup(
+        ansible_distro_api_client,
+        {
+            "name": pulp_repo.name,
+            "base_path": pulp_repo.name,
+            "repository_version": repository_version_latest.pulp_href,
+        },
+    )
+
+    # make sure the CV was indexed
+    dist_id = distro.pulp_href.split("/")[-2]
+    resp = galaxy_v3_default_search_api_client.list(limit=1000, distribution=[dist_id])
+    assert resp.meta.count == 1
```

Comment:
Weird that the generated client doesn't return the response with a direct `count` property like how it does for the Pulp clients.

Reply:
```(Epdb) resp.count
*** AttributeError: 'PaginatedCollectionVersionSearchListResponseList' object has no attribute 'count'
(Epdb) resp.meta.count
1
```